src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2119,7 +2119,7 @@ PlanRouterQuery(Query *originalQuery,
 	}
 	else if (replacePrunedQueryWithDummy)
 	{
-		List *workerNodeList = ActiveReadableNodeList();
+		List *workerNodeList = ActiveReadableWorkerNodeList();
```

Comment:
Self-note: we might consider adding only using localNode here as we have local execution.